README.md

```diff
@@ -1,2 +1,79 @@
-# token-wrap
-The SPL Token Wrap program and its clients
+# SPL Token Wrap Program
+
+[![GitHub Actions Workflow Status](https://img.shields.io/github/actions/workflow/status/solana-program/token-wrap/main.yml?logo=GitHub)](https://github.com/solana-program/token-wrap/actions/workflows/main.yml)
+
+This program enables the creation of "wrapped" versions of existing SPL tokens, facilitating interoperability
+between different token standards. If you are building an app with a mint/token and find yourself wishing you could take
+advantage of some of the latest features of a specific token program, this might be for you!
+
+## Features
+
+* **Bidirectional Wrapping:** Convert tokens between SPL Token and SPL Token 2022 standards in either direction,
+  including conversions between different SPL Token 2022 mints.
+* **SPL Token 2022 Extension Support:**  Preserve or add SPL Token 2022 extensions (like transfer fees, confidential
+  transfers, etc.) during the wrapping process.
+* **Transfer Hook Compatibility:**  Integrates with tokens that implement the SPL Transfer Hook interface,
+  enabling custom logic on token transfers.
+* **Multisignature Support:** Compatible with multisig signers for both wrapping and unwrapping operations.
+
+## How It Works
+
+It supports three primary operations:
```

Comment:
It might be worth pointing out that *everything* is permissionless, so it's possible for anyone to create wrapped mints, as long as they fund it